ecpy_hqc_legacy/instruments/drivers/visa/tektro_awg.py

```diff
@@ -556,6 +646,7 @@ def sampling_frequency(self, value):
     def running(self):
         """Run state getter method
         """
+        self.clear_output_buffer()
```

Comment:
Why so many calls to this method ? I guess it should be called in reopen_connection or in some error handling function.

Reply:
We discussed this. The AWG seems to not clear the output buffer. This must be an instrument bug. So it creates errors when you write read and you use an old answer stored in the buffer to set a parameter

Reply:
This is a known bug of the *CLS command not cleaning the output buffer. However this should only matter after recovering of an error as under normal operation the buffer should not get corrupted. Which is why this method should be refactored inside the clear method (not sure I have the name correctly and should at most be called only when opening the connection or even only when reopening. Otherwise it may mean that the driver is doing something incorrect and not properly reading the buffer at some point.

Reply:
Ok I will put it only in open and reopen, but I think this is what I did initially, and I still had this problem.